### DISCUSSION TITLE
Migrate from ActivityPub server to Mastodon while keeping domain and usernames.

### DISCUSSION THREAD

**Discussion by jcarnat** [mastodon/mastodon#24514] [Q&A]
Hi,

I already setup and used an ActivityPub server (not Mastodon) with several users for some time.
I would like to switch to using Mastodon while keeping the actual domain and usernames.
I read that users have private/public keys assigned and used by servers to "authenticate" the accounts.

If I configure a Mastodon server for my actual domain and create the corresponding usernames ; and then overwrite the generated users private/public keys  with the one grabbed from my actual ActivityPub server, will other servers accept my Mastodon users or will this fail ?

Thanks

PS: I plan to do this "offline" by not allowing Mastodon to access the Internet until the copy/paste is done and the old server down.

**Comment by nightpool** [comment]
this should basically work, but it might be a little tricky. the biggest thing to make sure of is that you'll need to migrate all of the "follow" relationships from the previous server so that your new mastodon server knows which remote servers to send posts to.

**Reply by jcarnat** [reply]
Ok, great. Yes I thought later on about transferring relationship too. Thank you.
